rdbms/src/main/java/org/spine3/server/storage/jdbc/entity/visibility/query/InsertVisibilityQuery.java

```diff
@@ -0,0 +1,124 @@
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.spine3.server.storage.jdbc.Sql.BuildingBlock.SEMICOLON;
+import static org.spine3.server.storage.jdbc.Sql.Query.INSERT_INTO;
+import static org.spine3.server.storage.jdbc.Sql.Query.VALUES;
+import static org.spine3.server.storage.jdbc.Sql.nPlaceholders;
+import static org.spine3.server.storage.jdbc.entity.visibility.table.VisibilityTable.TABLE_NAME;
+
+/**
+ * The query for creating a new record in the table storing
+ * the {@linkplain org.spine3.server.entity.Visibility entity visibility}.
+ *
+ * @author Dmytro Dashenkov.
+ */
+public class InsertVisibilityQuery extends WriteQuery {
+
+    private static final int COLUMN_COUNT = 3;
+    private static final String SQL = INSERT_INTO + TABLE_NAME +
+                                      VALUES + nPlaceholders(COLUMN_COUNT) + SEMICOLON;
+
+    private final String id;
+    private final Visibility entityStatus;
+
+    protected InsertVisibilityQuery(Builder builder) {
+        super(builder);
+        this.id = builder.id;
+        this.entityStatus = builder.entityStatus;
+    }
+
+    @Override
+    protected PreparedStatement prepareStatement(ConnectionWrapper connection) {
+        final PreparedStatement statement = super.prepareStatement(connection);
+        final boolean archived = entityStatus.getArchived();
+        final boolean deleted = entityStatus.getDeleted();
+        try {
+            statement.setString(TableColumn.ID.index, id);
+            statement.setBoolean(TableColumn.ARCHIVED.index, archived);
+            statement.setBoolean(TableColumn.DELETED.index, deleted);
+        } catch (SQLException e) {
+            logWriteError(id, e);
+            throw new DatabaseException(e);
+        }
+        return statement;
+    }
+
+    public static Builder newBuilder() {
+        final Builder builder = new Builder();
+        builder.setQuery(SQL);
+        return builder;
+    }
+
+    public static class Builder extends WriteQuery.Builder<Builder, InsertVisibilityQuery> {
+
+        private String id;
+        private Visibility entityStatus;
+
+        public Builder setVisibility(Visibility status) {
+            this.entityStatus = checkNotNull(status);
+            return getThis();
+        }
+
+        public Builder setId(Object id) {
+            checkNotNull(id);
+            final String stringId = Stringifiers.idToString(id);
+            this.id = stringId;
+            return getThis();
+        }
+
+        @Override
+        public InsertVisibilityQuery build() {
+            checkNotNull(id, "ID is not set.");
+            checkNotNull(entityStatus, "Entity status is not set.");
+            return new InsertVisibilityQuery(this);
+        }
+
+        @Override
+        protected Builder getThis() {
+            return this;
+        }
+    }
+
+    private enum TableColumn {
```

Comment:
The `TableColumn` cannot be used everywhere. The `QeuryParameter` `enum` describes the parameters of a specific query, which are not always the same (usually but necessary they are a subset of `TableColumn`s). If we rename something, I'd choose to use `QueryParameter`.